marshal/admin.go

```diff
@@ -0,0 +1,340 @@
+/*
+ * portal - marshal
+ *
+ * a library that implements an algorithm for doing consumer coordination within Kafka, rather
+ * than using Zookeeper or another external system.
+ *
+ */
+
+package marshal
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/dropbox/kafka/proto"
+)
+
+// Admins will wait a total of consumerReleaseClaimWaitTime
+// for paused consumers to release their claims on partitions.
+const (
```

Comment:
Should probably assert in NewAdmin that the pause duration >= consumerReleaseClaimWaitTime.